core/storage/file/gae_models_test.py

```diff
@@ -0,0 +1,229 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for core.storage.file.gae_models."""
+
+from core.platform import models
+from core.tests import test_utils
+import feconf
+
+(file_models, base_models) = models.Registry.import_models(
+    [models.NAMES.file, models.NAMES.base_model])
+
+
+class FileMetadataModelTest(test_utils.GenericTestBase):
+    """Tests the FileMetadataModel class."""
+
+    def test_get_new_id_raises_not_implemented_error(self):
+        with self.assertRaises(NotImplementedError):
+            file_models.FileMetadataModel.get_new_id('entity1')
+
+    def test_get_undeleted_with_two_undeleted_models_returns_both(self):
+        file_metadata_model1 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_metadata_model1.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model2 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file2.png')
+        file_metadata_model2.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model_list = (
+            file_models.FileMetadataModel.get_undeleted())
+        self.assertEqual(len(file_metadata_model_list), 2)
+        self.assertEqual(
+            file_metadata_model1, file_metadata_model_list[0])
+        self.assertEqual(
+            file_metadata_model2, file_metadata_model_list[1])
+
+    def test_get_undeleted_with_one_deleted_and_undeleted_model_returns_one(
+            self):
+        file_metadata_model1 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_metadata_model1.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model2 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file2.png')
+        file_metadata_model2.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model2.delete(
+            feconf.SYSTEM_COMMITTER_ID, 'Delete second file model')
+        file_metadata_model_list = (
+            file_models.FileMetadataModel.get_undeleted())
+        self.assertEqual(len(file_metadata_model_list), 1)
+        self.assertEqual(
+            file_metadata_model1, file_metadata_model_list[0])
+
+    def test_create_creates_model_with_correct_id_and_deleted_status_as_false(
+            self):
+        file_metadata_model1 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        self.assertEqual(file_metadata_model1.deleted, False)
```

Comment:
Nit (here & elsewhere for false/true cases), you can just use ``assertFalse()``/``assertTrue()``, I believe.